diff -ur gmp-2.0.2/Makefile.in gmp-2.0.2-hacked/Makefile.in
--- gmp-2.0.2/Makefile.in	Sun Feb 27 17:20:53 2000
+++ gmp-2.0.2-hacked/Makefile.in	Sun Feb 27 17:35:52 2000
@@ -97,7 +97,7 @@
 	    cp -p $$i/shared/* tmpdir/$$i; \
 	  done
 	cp -p shared/* tmpdir/
-	cd tmpdir; $(CC) -shared -Wl,-soname,libgmp.so.2 -o libgmp.so.2.0.2 *.o */*.o
+	cd tmpdir; $(CC) -shared -Wl,-G -Wl,-hlibgmp.so.2 -o libgmp.so.2.0.2 *.o */*.o
 	mv tmpdir/libgmp.so.2.0.2 .;ln -s libgmp.so.2.0.2 libgmp.so
 	rm -rf tmpdir
 
diff -ur gmp-2.0.2/mpn/sparc32/add_n.S gmp-2.0.2-hacked/mpn/sparc32/add_n.S
--- gmp-2.0.2/mpn/sparc32/add_n.S	Wed May  8 08:10:52 1996
+++ gmp-2.0.2-hacked/mpn/sparc32/add_n.S	Sun Feb 27 17:07:29 2000
@@ -29,7 +29,7 @@
 
 #include "sysdep.h"
 
-	.text
+.section	".text"
 	.align	4
 	.global	C_SYMBOL_NAME(__mpn_add_n)
 C_SYMBOL_NAME(__mpn_add_n):
diff -ur gmp-2.0.2/mpn/sparc32/addmul_1.S gmp-2.0.2-hacked/mpn/sparc32/addmul_1.S
--- gmp-2.0.2/mpn/sparc32/addmul_1.S	Wed May  8 08:10:52 1996
+++ gmp-2.0.2-hacked/mpn/sparc32/addmul_1.S	Sun Feb 27 17:07:41 2000
@@ -29,7 +29,7 @@
 
 #include "sysdep.h"
 
-.text
+.section	".text"
 	.align 4
 	.global	C_SYMBOL_NAME(__mpn_addmul_1)
 C_SYMBOL_NAME(__mpn_addmul_1):
diff -ur gmp-2.0.2/mpn/sparc32/lshift.S gmp-2.0.2-hacked/mpn/sparc32/lshift.S
--- gmp-2.0.2/mpn/sparc32/lshift.S	Wed May  8 08:10:53 1996
+++ gmp-2.0.2-hacked/mpn/sparc32/lshift.S	Sun Feb 27 17:07:49 2000
@@ -28,7 +28,7 @@
 
 #include "sysdep.h"
 
-	.text
+.section	".text"
 	.align	4
 	.global	C_SYMBOL_NAME(__mpn_lshift)
 C_SYMBOL_NAME(__mpn_lshift):
diff -ur gmp-2.0.2/mpn/sparc32/mul_1.S gmp-2.0.2-hacked/mpn/sparc32/mul_1.S
--- gmp-2.0.2/mpn/sparc32/mul_1.S	Wed May  8 08:10:52 1996
+++ gmp-2.0.2-hacked/mpn/sparc32/mul_1.S	Sun Feb 27 17:08:00 2000
@@ -91,7 +91,7 @@
 
 #include "sysdep.h"
 
-.text
+.section	".text"
 	.align 4
 	.global	C_SYMBOL_NAME(__mpn_mul_1)
 C_SYMBOL_NAME(__mpn_mul_1):
diff -ur gmp-2.0.2/mpn/sparc32/rshift.S gmp-2.0.2-hacked/mpn/sparc32/rshift.S
--- gmp-2.0.2/mpn/sparc32/rshift.S	Wed May  8 08:10:53 1996
+++ gmp-2.0.2-hacked/mpn/sparc32/rshift.S	Sun Feb 27 17:08:09 2000
@@ -28,7 +28,7 @@
 
 #include "sysdep.h"
 
-	.text
+.section	".text"
 	.align	4
 	.global	C_SYMBOL_NAME(__mpn_rshift)
 C_SYMBOL_NAME(__mpn_rshift):
diff -ur gmp-2.0.2/mpn/sparc32/sub_n.S gmp-2.0.2-hacked/mpn/sparc32/sub_n.S
--- gmp-2.0.2/mpn/sparc32/sub_n.S	Wed May  8 08:10:52 1996
+++ gmp-2.0.2-hacked/mpn/sparc32/sub_n.S	Sun Feb 27 17:08:18 2000
@@ -29,7 +29,7 @@
 
 #include "sysdep.h"
 
-	.text
+.section	".text"
 	.align	4
 	.global	C_SYMBOL_NAME(__mpn_sub_n)
 C_SYMBOL_NAME(__mpn_sub_n):
diff -ur gmp-2.0.2/mpn/sparc32/submul_1.S gmp-2.0.2-hacked/mpn/sparc32/submul_1.S
--- gmp-2.0.2/mpn/sparc32/submul_1.S	Wed May  8 08:10:52 1996
+++ gmp-2.0.2-hacked/mpn/sparc32/submul_1.S	Sun Feb 27 17:08:27 2000
@@ -29,7 +29,7 @@
 
 #include "sysdep.h"
 
-.text
+.section	".text"
 	.align 4
 	.global	C_SYMBOL_NAME(__mpn_submul_1)
 C_SYMBOL_NAME(__mpn_submul_1):
diff -ur gmp-2.0.2/mpn/sparc32/udiv_fp.S gmp-2.0.2-hacked/mpn/sparc32/udiv_fp.S
--- gmp-2.0.2/mpn/sparc32/udiv_fp.S	Sun Feb 27 17:20:53 2000
+++ gmp-2.0.2-hacked/mpn/sparc32/udiv_fp.S	Sun Feb 27 17:08:49 2000
@@ -33,17 +33,17 @@
 #undef ret	/* Kludge for glibc */
 
 #ifdef PIC
-	.text
+.section	".text"
 #else
-	.section .rodata,#alloc
+.section	".rodata"
 #endif
 	.align	8
-	.type two_to_32,@object
+	.type two_to_32,#object
 	.size two_to_32,8
 two_to_32:
 	.double	0r4294967296
 
-	.type two_to_31,@object
+	.type two_to_31,#object
 	.size two_to_31,8
 two_to_31:
 	.double	0r2147483648
diff -ur gmp-2.0.2/mpn/sparc32/udiv_nfp.S gmp-2.0.2-hacked/mpn/sparc32/udiv_nfp.S
--- gmp-2.0.2/mpn/sparc32/udiv_nfp.S	Wed May  8 08:10:52 1996
+++ gmp-2.0.2-hacked/mpn/sparc32/udiv_nfp.S	Sun Feb 27 17:09:01 2000
@@ -29,7 +29,7 @@
 
 #include "sysdep.h"
 
-	.text
+.section	".text"
 	.align 4
 	.global	C_SYMBOL_NAME(__udiv_qrnnd)
 C_SYMBOL_NAME(__udiv_qrnnd):
